bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller_test.go

```diff
@@ -508,7 +508,7 @@ func TestKubeadmConfigReconciler_Reconcile_GenerateCloudConfigData(t *testing.T)
 	k := &KubeadmConfigReconciler{
 		Client:              myclient,
 		SecretCachingClient: myclient,
-		Tracker:             remote.NewTestClusterCacheTracker(logr.New(log.NullLogSink{}), myclient, myclient.Scheme(), client.ObjectKey{Name: cluster.Name, Namespace: cluster.Namespace}),
+		Tracker:             remote.NewTestClusterCacheTracker(logr.New(log.NullLogSink{}), myclient, myclient, myclient.Scheme(), client.ObjectKey{Name: cluster.Name, Namespace: cluster.Namespace}),
```

Comment:
I added a new parameter to `NewTestClusterCacheTracker`. Generally, I don't like how all unit tests mix the management cluster's k8s client with the workload cluster's client ("remote") and hence store all objects within the same fakeClient's storage. This can be confusing for developers since you don't really know whether to expect something in the management or workload cluster. So for the test I added, there's now a split between those two types of client → bootstrap data in the management cluster, bootstrap token secret in the workload cluster. With this, we also test that CAPA actually puts objects in the correct cluster. However, I didn't adapt all tests because that would have blown up the PR scope.